src/lib/models/group-events/get-group-for-event.ts

```diff
@@ -0,0 +1,15 @@
+export const getGroupForEvent = (
+  event: HistoryEventWithId,
+  groups: CompactEventGroups,
+): CompactEventGroup => {
+  const eventId = event.id;
+
+  for (const group of groups) {
+    if (eventId === group.id) return group;
+    for (const ev of group.events.values()) {
+      if (ev.id === event.id) {
+        return group;
```

Comment:
A test or two to verify happy/sad paths on this would probably go a long way.

Reply:
As we discussed, it had a test, but I added an additional test too.